File: modules/crypto-core/src/main/java/com/shterneregen/securelan/crypto/util/CryptoExceptions.java
package com.shterneregen.securelan.crypto.util;

public final class CryptoExceptions {
    private CryptoExceptions() {
    }

    public static IllegalArgumentException missing(String name) {
        return new IllegalArgumentException(name + " must not be null");
    }

    public static IllegalArgumentException empty(String name) {
        return new IllegalArgumentException(name + " must not be empty");
    }

    public static IllegalStateException failed(String action, Exception cause) {
        return new IllegalStateException("Failed to " + action, cause);
    }
}

File: modules/crypto-core/src/main/kotlin/com/shterneregen/securelan/crypto/util/CryptoExceptions.kt
package com.shterneregen.securelan.crypto.util

class CryptoExceptions private constructor() {
    companion object {
        @JvmStatic
        fun missing(name: String): IllegalArgumentException = IllegalArgumentException("$name must not be null")

        @JvmStatic
        fun empty(name: String): IllegalArgumentException = IllegalArgumentException("$name must not be empty")

        @JvmStatic
        fun failed(action: String, cause: Exception): IllegalStateException = IllegalStateException("Failed to $action", cause)
    }
}
